### PR TITLE
enhancement(remap): Pretty print functions list in VRL REPL

### PR DESCRIPTION
**Pull request by lucperkins**
This is a first stab at something that I believe could help users with VRL. If you run `functions` in the REPL you get a big table that looks like this:

```
+-------------------+-----------------------------------------------+-------------------------+
| assert            | ceil                                          | compact                 |
| contains          | del                                           | downcase                |
| encode_json       | ends_with                                     | exists                  |
| parse_regex       | parse_regex_all                               | flatten                 |
| floor             | format_number                                 | format_timestamp        |
| includes          | ip_cidr_contains                              | ip_subnet               |
| ip_to_ipv6        | ipv6_to_ipv4                                  | is_nullish              |
| log               | md5                                           | merge                   |
| now               | ok                                            | only_fields             |
| parse_aws_alb_log | parse_aws_cloudwatch_log_subscription_message | parse_aws_vpc_flow_log  |
| parse_duration    | parse_grok                                    | parse_json              |
| parse_key_value   | parse_syslog                                  | parse_timestamp         |
| parse_url         | match                                         | redact                  |
| replace           | round                                         | sha1                    |
| sha2              | sha3                                          | slice                   |
| split             | starts_with                                   | strip_ansi_escape_codes |
| strip_whitespace  | to_bool                                       | to_float                |
| to_int            | to_syslog_level                               | to_syslog_severity      |
| to_string         | to_timestamp                                  | tokenize                |
| truncate          | upcase                                        | uuid_v4                 |
+-------------------+-----------------------------------------------+-------------------------+
```

There are other possibilities we could explore here but this at least provides a little helper for moments when you're thinking "wait, what's that function called again?" or "wait, which functions are in this version?" Future work could allow for pulling up "docs" for specific functions, e.g. `func format_timestamp` gets you a description, parameters, fallibility, docs URL, etc.

Also, the choice of three cells in a row was completely arbitrary, so please advise if that's too many/few. Unfortunately, names like `parse_aws_cloudwatch_log_subscription_message` do make the aesthetics a bit tricky.

### CODE REVIEW

**Comment by JeanMertz** [comment]
I’d prefer it if we move any help-like capabilities under the “help” command.

e.g. `help` prints a generic help message and also prints the availability of `help function` to get a list of all functions and then later also support `help function parse_json` to get a description and example of that function.

in order to support that last part we could expand our function trait so that we are always required to add an example and a description to any function we create for VRL itself to compile. This would then also allow us down the line to programmatically update our Cue docs based on the implementation itself to keep all of it in sync.

This last part is of course orthogonal to this PR, but it’s an idea that came to mind based on it.

**Comment by lucperkins** [comment]
@JeanMertz Okay, that sounds good to me. FYI I've laid out a handful of pretty basic ideas in #5949.

**Review by FungusHumungus — APPROVED** [review]

File: lib/remap-cli/src/repl.rs [hunk 1]
```diff
@@ -8,12 +9,13 @@ use rustyline::validate::{self, MatchingBracketValidator, ValidationResult, Vali
...
 use std::borrow::Cow::{self, Borrowed, Owned};
 
-const HELP_TEXT: &str = "
+const HELP_TEXT: &str = r#"
 VRL REPL commands:
-  next      Load the next object or create a new one
-  prev      Load the previous object
-  exit      Terminate the program
-";
+  help function  Display a list of currently available VRL functions (aliases: ["help func", "help f"])
```
Comment (JeanMertz) [inline lib/remap-cli/src/repl.rs:14]: Maybe make `help functions` work as well, for consistency’s sake.

**Review by JeanMertz — APPROVED** [review]
Sweet!

**Pull request merged**